Move the filters to the right side

zoglo opened the pull request:
### Description

* Implements #6021  / fixes #6021 / supersedes #6146

#### What does this PR do

* Previously, filters had no labels, which made them not accessible. Now proper labels are added.
* With many filters, the previous layout looked messy and inconsistent, and horizontal space was limited. The new layout switches it to be vertical, ensuring a clean, readable, and accessible appearance regardless of the number of filters.

#### Disclaimer

* This is not yet ready for mobile, I would maybe need https://github.com/contao/contao/pull/8618 to improve it 

#### Images

<img width="1883" height="1160" alt="image" src="https://github.com/user-attachments/assets/a1bc7e54-5a73-468b-bc2b-6b7dfb8522bb" />
<img width="1885" height="1198" alt="image" src="https://github.com/user-attachments/assets/0027f442-42ed-42c5-a2a3-885abe3ece8e" />
<img width="1883" height="1068" alt="image" src="https://github.com/user-attachments/assets/d881a60b-7cb5-4992-b648-a69d6e4a4c61" />
<img width="1887" height="1136" alt="image" src="https://github.com/user-attachments/assets/6c944334-31af-4991-b7c3-2a6de28bc51f" />
<img width="1882" height="1247" alt="image" src="https://github.com/user-attachments/assets/f19dddc9-9d1b-43c9-a23e-ac40ff9b82c8" />



#### Different breakpoints
<img width="374" height="1276" alt="image" src="https://github.com/user-attachments/assets/07aace97-469d-499b-a282-e2c6e2321884" />
<img width="628" height="1306" alt="image" src="https://github.com/user-attachments/assets/ab9bc483-d0b0-4e7a-9910-bc4db2bb11f8" />
<img width="778" height="1188" alt="image" src="https://github.com/user-attachments/assets/10013348-e786-4595-b6c2-e8ff9e704bdd" />
<img width="977" height="1185" alt="image" src="https://github.com/user-attachments/assets/a4b7cff1-75c5-46bb-aea6-a1f2efc5ee93" />
<img width="1235" height="1167" alt="image" src="https://github.com/user-attachments/assets/8e382c56-67de-4ae1-b326-99dc06a9a475" />

zoglo committed:
Move the filters to the right side

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
CS

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Add the missing labels to the search

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Make it responsive

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Also use a label for tl_sort

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Do not grid the tree-view if conten-filter does not exist

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Consider versions + adjust filter buttons

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Merge branch '5.x' into feature/right-filter

fritzmg commented:
Minor nitpick: I think the bottom padding of the form buttons could use the same padding as left, top & right:

<img width="225" src="https://github.com/user-attachments/assets/37a111e8-4497-4f18-a905-5f9ca9355a73">

fritzmg commented on core-bundle/contao/themes/flexible/styles/base/responsive.pcss:
```diff
@@ -119,9 +119,9 @@
+    /*.tl_filter strong {
         display: none;
-    }
+    }*/
```
This leftover can probably be removed?

zoglo replied:
See 796f2bc5b21decfd21620c5dae034940fc88712c

zoglo committed:
Clean leftover + adjust padding

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

Toflar reviewed (dismissed):
Love it! ❤️ 
When testing, I noticed that the order of the panel layout doesn't make as much sense anymore now. For example, I would probably put the sorting options last and the search I would likely put first. However, this can also be done in a separate PR because it's not related to the styling and can also be influenced individually in extensions.

leofeyer commented:
I like this PR a lot, but it decreases usability on mobile devices. Since every field now has a label above, the filter area is now much taller and I need to scroll more than before to apply a filter

leofeyer commented:
I don't understand. How does the `toggle-state-controller` fix the decreased usability?

leofeyer commented:
But that does not solve the problem that the filters are much taller now, does it?

leofeyer commented:
But I cannot merge a PR that worsens usability.

zoglo commented:
Thinking about it right now, this would need more redesign so I am gonna set it to a draft.

zoglo commented:
@leofeyer what about moving the filters to the bottom as the content matters first:

https://github.com/contao/contao/pull/8850#pullrequestreview-3224064001

Then you would not need to scroll 1200px to get to the list?

zoglo committed:
Merge branch '5.x' into feature/right-filter

stefansl commented:
<img width="1903" height="1482" alt="Bildschirmfoto 2025-09-27 um 11 16 26 min" src="https://github.com/user-attachments/assets/f715e7a4-5dd1-4ced-b821-e9df12d3af28" />
<img width="1895" height="1446" alt="Bildschirmfoto 2025-09-27 um 11 16 32 min" src="https://github.com/user-attachments/assets/28278dd3-d645-4c1e-824a-f28e09dec0ee" />

What about this approach from Shopware? By default the filters are hidden?

zoglo committed:
Merge branch '5.x' into feature/right-filter

zoglo committed:
Move the submit panel under the formbody

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Add drawer styles (mobile)

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Rebuild the assets

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Fix the comment and add the assets

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Merge branch '5.x' into feature/right-filter

zoglo committed:
Fix merge conflict

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Wrap the content within `show-all`

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Make use of grid instead of template changes

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Merge branch '5.x' into feature/right-filter

zoglo committed:
CS

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Finish mobile UX

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
CS

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo committed:
Remove obsolete translation

Signed-off-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo commented:
> What about this approach from Shopware? By default the filters are hidden?

@stefansl for now it's about implementing the necessary layout and behavior before adding other changes. A UX design has already been created for this PR and may be implemented iteratively.

zoglo committed:
Merge branch '5.x' into feature/right-filter

m-vo commented:
In the view for reduced widths, the `Toggle filter` is not working for me. Nothing opens - instead, when scrolling fully down, there is a margin at the end of the document (below the entire HTML element).

zoglo committed:
Introduce a filter controller to count active filters

zoglo committed:
Assets

zoglo committed:
Lint

zoglo commented:
> when scrolling completely down on a vertically large view, the buttons move up a bit (because they are "fixed" to the main part)

This is another bug, see https://github.com/contao/contao/issues/6630 for reference

zoglo committed:
Also include DC_Folder

m-vo commented on core-bundle/contao/templates/backend/be_main.html5:
```diff
@@ -106,14 +116,14 @@ $this->containerAttributes = $this
 
       <?php $this->block('main'); ?>
-        <main id="main" aria-labelledby="main_headline">
+        <main id="main"<?= $this->mainAttributes ?>>
```
```suggestion
        <main<?= $this->mainAttributes ?>>
```

m-vo commented on core-bundle/contao/templates/backend/be_main.html5:
```diff
@@ -25,6 +25,16 @@ $this->containerAttributes = $this
 ;
 
+$this->mainAttributes = $this
+    ->attrs($this->attributes)
```
```suggestion
    ->attrs()
```

What is `$this->attributes`? It's not present in the Twig version of this template.

m-vo commented on core-bundle/contao/templates/twig/backend/data_container/table/menu/filter.html.twig:
This should probably be a semantic headline/fieldset legend (see below).

m-vo commented on core-bundle/contao/templates/twig/be_main.html.twig:
```diff
@@ -22,6 +22,15 @@
+    .set('data-action', 'keydown.esc@document->contao--toggle-state#close')
+    .set('data-contao--toggle-state-active-class', 'filter-active')
+    .mergeWith(mainAttributes|default)
+-%}
```
(I previously had erroneously suggested using snake_case for the variables, but this is a surrogate template, so we don't apply the Twig CS here but have to keep compatibility with the PHP template.)

zoglo committed:
Apply suggestions from code review

Co-authored-by: M. Vondano <m-vo@users.noreply.github.com>

zoglo committed:
Do not set the id on main

Co-authored-by: M. Vondano <m-vo@users.noreply.github.com>

zoglo committed:
Update targeting the wrong select

zoglo committed:
Update label for tl_sort

aschempp commented:
The order of filter options seems to be changed? Does not happen in 5.x

<img width="304" height="140" alt="Bildschirmfoto 2025-12-14 um 19 53 22" src="https://github.com/user-attachments/assets/a4f3924e-e9c0-4884-92f8-f514b53bed48" />

fritzmg reviewed (commented):
~There is a double `id="main"` in the HTML output:~

```html
<main id="main" id="main" …
```

// edit: nvm, that's in the main branch

aschempp commented:
An empty preview link list needs some fix with the message:

<img width="1203" height="373" alt="Bildschirmfoto 2025-12-14 um 20 27 44" src="https://github.com/user-attachments/assets/9337d504-1400-4ba6-a192-6ba7c38c80fd" />

zoglo committed:
Apply oopsie from code review

Co-authored-by: Fritz Michael Gschwantner <fmg@inspiredminds.at>

zoglo commented:
> My member list has more than 30 entries and looks like this. Current 5.x does not have that issue.
> <img alt="Bildschirmfoto 2025-12-14 um 19 48 16" width="1227" height="438" src="https://private-user-images.githubusercontent.com/1073273/526324624-884367ae-dc0f-4eca-aeff-22eb5919c696.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU3NTA5NjQsIm5iZiI6MTc2NTc1MDY2NCwicGF0aCI6Ii8xMDczMjczLzUyNjMyNDYyNC04ODQzNjdhZS1kYzBmLTRlY2EtYWVmZi0yMmViNTkxOWM2OTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMTRUMjIxNzQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTRhMzBkYTcwNzM2ZmZjY2RhNjM2MjBmMmQ4MzZjYjQ5ZWZlMTVkZGRhNTAyZTUwOWIzM2JmZGNlMjVlZDNjZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.oglsvpO8KXRUmSnlNSnM_LlAbf8XyqPFb02CSC9ccCw">



> The layout in the modal also needs some adjustment, e.g. when picking a file:
> <img alt="" width="571" src="https://private-user-images.githubusercontent.com/4970961/526326884-cb27748e-d01d-469e-bd84-b0f96ec4da9a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU3NTA5NjQsIm5iZiI6MTc2NTc1MDY2NCwicGF0aCI6Ii80OTcwOTYxLzUyNjMyNjg4NC1jYjI3NzQ4ZS1kMDFkLTQ2OWUtYmQ4NC1iMGY5NmVjNGRhOWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMTRUMjIxNzQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjAzM2Q4NzcxMjQxNmFjNGY4OWE5MmMyNTZjNTJmNTYzNzRkZDMzZTcyNmI5YThlYjIxMjE3Yjc5M2QwNzI0ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.U_uKHzRRk3_BCiUPXNVpolHCRxGERpx1sqYWyFkDrh8">

These are due to c3b307f82106bb1f3add13cbc2203df7d5cb7197 (changed to css only for the other PR to not cause a bazillion conflicts, will revert that now and go with the proposed layout changes)

zoglo committed:
Revert c3b307f82106bb1f3add13cbc2203df7d5cb7197 and auto-overflow long table entries within the content

zoglo committed:
Rebuild the assets

zoglo committed:
Merge remote-tracking branch 'zoglo/feature/right-filter' into feature/right-filter

zoglo committed:
Also handle popups

zoglo committed:
Also handle popups again

zoglo commented:
> My member list has more than 30 entries and looks like this. Current 5.x does not have that issue.
> <img alt="Bildschirmfoto 2025-12-14 um 19 48 16" width="1227" height="438" src="https://private-user-images.githubusercontent.com/1073273/526324624-884367ae-dc0f-4eca-aeff-22eb5919c696.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU3NTMyMDgsIm5iZiI6MTc2NTc1MjkwOCwicGF0aCI6Ii8xMDczMjczLzUyNjMyNDYyNC04ODQzNjdhZS1kYzBmLTRlY2EtYWVmZi0yMmViNTkxOWM2OTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMTRUMjI1NTA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDg3MjllZGVjZDQ3OWMxMjM1YTJlOGZhN2RkYmEyYjJhNzUzNTk5ODU5ZmQ0MzBmYTc3OGI0MGE3YjAyYjZjNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.c86K4WZRmHjsHum_Sr_IT0d9U6vLZaT72Jjk9r3aJ8A">

Has been fixed, please check again (partly reverted the change explained in https://github.com/contao/contao/pull/8850#issuecomment-3652320627)

> There is a double `id="main"` in the HTML output:
> 
> ```
> <main id="main" id="main" …
> ```

fixed

> The layout in the modal also needs some adjustment, e.g. when picking a file:
> <img alt="" width="571" src="https://private-user-images.githubusercontent.com/4970961/526326884-cb27748e-d01d-469e-bd84-b0f96ec4da9a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU3NTMyMDksIm5iZiI6MTc2NTc1MjkwOSwicGF0aCI6Ii80OTcwOTYxLzUyNjMyNjg4NC1jYjI3NzQ4ZS1kMDFkLTQ2OWUtYmQ4NC1iMGY5NmVjNGRhOWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMTRUMjI1NTA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTUxYzQwMGFjMjllYTI2NTYyNWQyZTBhYmZlNzkyNmRhMzBkMjgxMDdlYzU4MWVhMGE5NjQyMjkwMDFiZTViYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.OKqo074VuzS1nQk5rvohQz-pMqCpZysVyfK70mZwfvQ">

That should also be fixed in the lastest commits, please review again

> An empty preview link list needs some fix with the message:
> <img alt="Bildschirmfoto 2025-12-14 um 20 27 44" width="1203" height="373" src="https://private-user-images.githubusercontent.com/1073273/526327552-9337d504-1400-4ba6-a192-6ba7c38c80fd.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU3NTMyMDgsIm5iZiI6MTc2NTc1MjkwOCwicGF0aCI6Ii8xMDczMjczLzUyNjMyNzU1Mi05MzM3ZDUwNC0xNDAwLTRiYTYtYTE5Mi02YmE3YzM4YzgwZmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMTRUMjI1NTA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWE1MjYwYzdlNTcwNmNmZmZmZjk2ODI4NTM3MTc1NmU2ZjQ0MzhjZTg2Y2Q5ZmRkNzI5Y2FjM2ZlNGM3ZjNhNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.w9pqr822k0glkDTz8ZH0T_iUrjLZHxbBCuvBpzwEy8s">

fixed as well :)

zoglo commented:
> The order of filter options seems to be changed? Does not happen in 5.x
> <img alt="Bildschirmfoto 2025-12-14 um 19 53 22" width="304" height="140" src="https://private-user-images.githubusercontent.com/1073273/526325018-a4f3924e-e9c0-4884-92f8-f514b53bed48.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU3NTMyMDgsIm5iZiI6MTc2NTc1MjkwOCwicGF0aCI6Ii8xMDczMjczLzUyNjMyNTAxOC1hNGYzOTI0ZS1lOWMwLTQ4ODQtOTJmOC1mNTE0YjUzYmVkNDgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMTRUMjI1NTA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmJjMTZmOTBhMDNmOWIyN2U0MzVhMDhjMmI0ZGI2ZDcxZWMyMjFmYTFmOGVhMTYxMmQ1MmM2MGM1NmMwN2QzMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Xe0pK55wxI1uFHil0Dmml8AlFJN01cfUim7Vdy0fE6U">

Unsure but I feel this is unrelated to this PR? Can you check outside of it?

zoglo committed:
Only overflow in specific tables

aschempp commented:
Is it possible to scroll the filter panel on mobile independencly of the back ground? If there are a lot of records, the panel as a long white context to scroll.
<img width="634" height="871" alt="Bildschirmfoto 2025-12-15 um 07 01 55" src="https://github.com/user-attachments/assets/cde2ad1a-84d5-4903-85f0-8dce17057382" />

Btw. LOVE the filter count in the global operations on mobile! I played around with the colors a bit, not sure if it would make sense to keep the "active" color? Looks better on dark to me, but not sure about light. Maybe with bold?

<img width="309" height="114" alt="Bildschirmfoto 2025-12-15 um 07 07 28" src="https://github.com/user-attachments/assets/b26df33b-1162-4adc-8678-7a42996c0cb2" />
<img width="336" height="101" alt="Bildschirmfoto 2025-12-15 um 07 07 32" src="https://github.com/user-attachments/assets/634c2496-47c4-4bdb-8e30-bfede0f302f6" />
<img width="372" height="106" alt="Bildschirmfoto 2025-12-15 um 07 08 18" src="https://github.com/user-attachments/assets/06085ba6-e578-433c-aa75-1e435e59cafb" />

aschempp commented:
The pagination and the apply/reset button sections seem to use two different colors. Is this intended or maybe a generic flaw in the back end (panels and submit container having two separate colors for no reason)?
<img width="665" height="138" alt="Bildschirmfoto 2025-12-15 um 07 17 14" src="https://github.com/user-attachments/assets/5e7dc8cd-7db7-4f52-b899-d934e852997d" />

zoglo committed:
Merge branch '5.x' into feature/right-filter

zoglo committed:
Fix missing class after merge

zoglo committed:
Make the filter the last operation

zoglo committed:
Always append the filter button

zoglo committed:
Remove the old popup resets and apply height 100%

zoglo committed:
Apply the sticky observer to the submit panel

zoglo committed:
Decouple the filter layout from the main content

zoglo committed:
Add an overflow to the message

zoglo committed:
Revert unrelated change

zoglo committed:
Move the filter toggle back to the global operations buider

zoglo committed:
Apply the filter buttons directly within DC_Table and DC_Folder

zoglo committed:
Ucomment the test again

zoglo committed:
Revert the popup changes + change the filter width

zoglo committed:
Allow the filter panel to always be sticky

zoglo committed:
Fix the z-index issues for the mobile filter panel

zoglo committed:
Show side-drawer filters till 1280px

zoglo committed:
Rebuild the assets

m-vo committed:
Use a fieldset to group elements and set a label for selects consistently

* use a fieldset to group multiple fields and always set a label for selects

Signed-off-by: M. Vondano <m-vo@users.noreply.github.com>

* Rebuild the assets

---------

Signed-off-by: M. Vondano <m-vo@users.noreply.github.com>
Co-authored-by: Sebastian Zoglowek <55794780+zoglo@users.noreply.github.com>

zoglo commented:
> What about [zoglo#5](https://github.com/zoglo/contao/pull/5)? This way the templates are consistent, we ensure to always set a label for the select and make the groups of things semantically correct. I think we should not postpone this.

Yup :)

aschempp commented:
> What about [zoglo#5](https://github.com/zoglo/contao/pull/5)? This way the templates are consistent, we ensure to always set a label for the select and make the groups of things semantically correct. I think we should not postpone this.

I'm not sure about this. `fieldset` is meant to group fields that are related to each other (like radio buttons, or number, name, date and CVS of a credit card), but the filter fields are not related to each other at all, except that they all are filter fields? https://www.w3.org/WAI/tutorials/forms/grouping/ has some examples about this, but none are about "unrelated" fields.

zoglo commented:
Andy does have a point here. I would most likely revert the fieldsets for now and use strongs and re-evaluate this in another PR @m-vo. We could however open an issue, would you be okay with that?

zoglo committed:
Merge branch '5.x' into feature/right-filter

zoglo committed:
Limit the max width of filters

zoglo committed:
Allow document click to close the filters

zoglo committed:
Add extra margin on the fixed filter panel

zoglo commented:
> Is it intended that the drawer now has ~70% of my screen with?

changed in 79f24ea627f8f69ef8b286b74730a749ef1e603e

> I would prefer if it is possible to close the filter panel by clicking outside on the overlay

Done in 7ce11e121451cd399c2a4f2ff28d7629c091bc53

> It's hard to see - but if the filter reaches to the bottom of the viewport while being sticky (and the submit container being sticky as well) then there is a border-radius missing when reaching the bottom of the viewport.

Fixed in b31d212c4e0e9dc90487b079d1e374d0327517c1 (also something that @aschempp mentioned)
<img width="387" height="424" alt="grafik" src="https://github.com/user-attachments/assets/aca1a17d-e38f-4235-a109-3ba274b0615b" />

leofeyer committed:
Merge branch '5.x' into feature/right-filter

leofeyer committed:
Rebuild the assets

leofeyer commented on core-bundle/contao/templates/twig/be_main.html.twig:
```diff
@@ -22,6 +22,16 @@
 
+{% set mainAttributes = attrs()
+    .set('id', 'main')
+    .addClass('be_main')
```
```suggestion
```

zoglo committed:
Add border-top + adjust the label size

zoglo committed:
Remove `be_main` class from mainAttributes

Co-authored-by: Leo Feyer <1192057+leofeyer@users.noreply.github.com>

zoglo committed:
Rebuild the assets

leofeyer commented:
Also, I would probably reduce the width from 280px to 240px.

<img width="669" height="737" alt="" src="https://github.com/user-attachments/assets/625b0c4b-89e1-465b-b225-ffb3ada236e5" />

zoglo commented:
> Also, I would probably reduce the width from 280px to 240px.

Unsure about that with other languages and the buttons, we do have the space 🤔. Are you referring to the same width as the left navigation?

zoglo committed:
Merge branch '5.x' into feature/right-filter

zoglo committed:
Reduce the filter width in `dwarf` mode

zoglo commented:
> Yes. And the goal is still to keep the visual focus on the main column, especially with a maximum back end width of 1280px.

I reduced it for the small percentage of people that would use this width 🤔 (only applied when the option is active)
f1f4b774d6bedd9a4284e8c9871581450ba5e329

zoglo committed:
Do not overflow the container till 1280px (filter)

zoglo committed:
Add `testAddFilterButton`

leofeyer committed:
Fix some minor styling issues

leofeyer committed:
Replace "Term" with "Keyword"

leofeyer committed:
Fix the tooltip of the filters button

leofeyer committed:
Fix the unit tests
